### ISSUE TITLE
netty http HttpObjectAggregator handler not support no body response for example http head request

### ISSUE THREAD

**Issue by gavenpeng** [netty/netty#7130]
### Actual behavior

HttpResponseDecoder decode only one message(response line and header) ,pass to HttpObjectAggregator , HttpObjectAggregator process as start message and wait content message
but no body,so HttpObjectAggregator can not do finishAggregation, so  HttpObjectAggregator out list 
size is 0 ,can not fire to my handler

### Netty version
netty-4.1.9

**Comment by normanmaurer** [comment]
Just use the `HttpClientCodec` and it will be handled correctly.

**Comment by gavenpeng** [comment]
thanks

**Comment by hamdiallam** [comment]
Hey @normanmaurer, can you elaborate on what you mean by using the HttpClientCodec with the HttpObjectAggregator? Both handlers are in the channel pipeline and the same problem arises. The HttpObectAggregator will throw the exception when it sees the header

**Comment by chrisvest** [comment]
@hamdiallam You place the `HttpClientCodec` before the `HttpObjectAggregator` in the channel pipeline. The `HttpObjectAggregator` javadocs has an example.

**Comment by pylight** [comment]
Having the same issue with netty 4.1.94.Final and a client-side HTTP HEAD request with a netty pipeline that uses
```
 pipeline.addLast(new HttpClientCodec());
 pipeline.addLast(new` HttpObjectAggregator(8192);
```

The `HttpObjectAggregator` throws **TooLongHttpContentException** even for the HEAD requests. I also found no information about `HttpClientCodec` in the javadoc of `HttpObjectAggregator`. I'm not quite sure but I thought that this code worked properly in earlier netty versions. Is there a way in the newer versions to use `HttpObjectAggregator` with HEAD requests and ignore the content-length?

**Comment by normanmaurer** [comment]
@pylight please open a new issue with a reproducer

**Comment by pylight** [comment]
FYI: I've looked into creating a reproducer today but it seems that most public websites (wanted to use something like "github.com:443" as server endpoint) don't send the Content-Length header in case of HEAD requests or they send "Content-Length: 0". 
Besides chunked transfer ending is often used in which case there's no Content-Length header in the response as well.
 
See also this related stackoverflow discussion where it is stated that the header MAY be omitted for HEAD responses:
https://stackoverflow.com/questions/3854842/content-length-header-with-head-requests

So I decided to change the server on my side as well to not include the header for HEAD responses. That's sufficient to not experience this issue anymore.
